test/unit/gateways/ebanx_test.rb

```diff
@@ -54,6 +54,17 @@ def test_successful_purchase_with_soft_descriptor
     assert_success response
   end
 
+  def test_successful_purchase_with_merchant_payment_code
+    response = stub_comms(@gateway, :ssl_request) do
+      @gateway.purchase(@amount, @credit_card, @options.merge(merchant_payment_code: '2'))
+    end.check_request do |_method, _endpoint, data, _headers|
+      assert_match %r{"merchant_payment_code\":\"2\"}, data
+      assert_match %r{"order_number\":\"2\"}, data
```

Comment:
The `order_number` assertion will need to change to account for the fact that we are still mapping it to `order_id`